packages/netlify-cms-core/src/actions/mediaLibrary.ts

```diff
@@ -489,18 +489,20 @@ export async function getMediaDisplayURL(
     // url loading had an error
     url = null;
   } else {
-    if (!displayURLState.get('isFetching')) {
-      // load display url
-      dispatch(loadMediaDisplayURL(file));
```

Comment:
`loadMediaDisplayURL` can actually finish and update the state before we start `waitUntilWithTimeout`